custom-api/acme_server.py

```diff
@@ -0,0 +1,289 @@
+from flask import Flask, request, jsonify, render_template
+from enum import Enum
+from urllib.parse import urlparse
+import os
+import requests
+import json
+import random
+import subprocess
+import threading
+import time
+
+# Databricks SDK
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.sql import *
+
+# Initialize Databricks SDK. 
+# By default, this authenticates using the DATABRICKS_HOST and DATABRICKS_TOKEN environment variables
+# initialized by setup.sh.
+w = WorkspaceClient()
+
+warehouse_id = os.environ.get("DATABRICKS_WAREHOUSE_ID")
+
+app = Flask(__name__)
+
+#
+# Store the parameterized SQL statements used to read/write data to Databricks SQL.
+#
+class SqlStatement(Enum):
+    LIST_STORES = 1
+    LIST_SALES = 2
+    INSERT_SALE = 3
+
+sql_statements = {
+     SqlStatement.LIST_STORES: """
+            select
+                id,
+                name, 
+                manager,
+                employee_count
+                city,
+                state
+            from 
+                hive_metastore.acme_demo.stores
+        """,
+    SqlStatement.LIST_SALES: """
+            select 
+                date,
+                id,
+                store_id,
+                item_id,
+                quantity,
+                price
+            from 
+                hive_metastore.acme_demo.sales
+            where
+                store_id = :store_id
+            order by
+                date desc
+        """,
+    SqlStatement.INSERT_SALE: """
+            INSERT INTO
+              hive_metastore.acme_demo.sales (
+                date,
+                id,
+                store_id,
+                item_id,
+                price,
+                quantity
+              )
+            VALUES
+              (
+                :sold_date,
+                :sale_id, 
+                :store_id,
+                :item_id,
+                :price,
+                :quantity
+              )
+        """
+}
+
+#
+# Web Endpoints
+#
+@app.route("/")
+def index():
+    return render_template("stores.html")
+
+@app.route("/stores/<string:store_id>")
+def customer(store_id):
+    return render_template("store.html", store_id=store_id)
+
+@app.route("/newsale")
+def newSale():
+    store_id = request.args.get('store_id')
+    return render_template("new_sale.html", store_id=store_id)
+
+#
+# Custom API Endpoints
+#
+
+#
+# Endpoint to list store information.
+#
+# It takes no arguments and invokes the Statement Execution API in synchronous mode, returning store data results in
+# JSON format. If the request to Databricks SQL takes longer than 50 seconds, the execution will be cancelled and no
+# data is returned.
+#
+@app.route("/api/1.0/stores", methods=["GET"])
+def list_stores():
+    statement_response = w.statement_execution.execute_statement(
+        statement = sql_statements[SqlStatement.LIST_STORES],
+        wait_timeout = "50s",
+        on_wait_timeout = TimeoutAction.CANCEL,
```

Comment:
Got this error. 
```
[2023-09-20 22:19:18,347] ERROR in app: Exception on /api/1.0/stores [GET]
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/varun.sharma/databricks/dbsql-rest-api/custom-api/acme_server.py", line 114, in list_stores
    on_wait_timeout = TimeoutAction.CANCEL,
                      ^^^^^^^^^^^^^
NameError: name 'TimeoutAction' is not defined
```

I think it should be ExecuteStatementRequestOnWaitTimeout instead as changing code to that helped.

Reply:
Seems like a regression in the SDK.